api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -74,5 +90,22 @@
    *
    * @return this for method chaining
    */
-  OverwriteFiles validateAddedFiles();
+  OverwriteFiles validateAddedFilesMatchRowFilter();
```

Comment:
Since a row filter is passed to `validateNoConflictingAppends`, I think this should be `validateAddedFilesMatchOverwriteFilter` instead.